File: app/src/main/java/dev/ragnarok/fenrir/api/model/response/AttachmentsHistoryResponse.java
package dev.ragnarok.fenrir.api.model.response;

import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import dev.ragnarok.fenrir.api.model.VKApiAttachments;
import dev.ragnarok.fenrir.api.model.VKApiCommunity;
import dev.ragnarok.fenrir.api.model.VKApiUser;

public class AttachmentsHistoryResponse {

    @Nullable
    @SerializedName("items")
    public List<One> items;

    @Nullable
    @SerializedName("next_from")
    public String next_from;

    @Nullable
    @SerializedName("profiles")
    public List<VKApiUser> profiles;

    @Nullable
    @SerializedName("groups")
    public List<VKApiCommunity> groups;

    public static class One {

        @SerializedName("message_id")
        public int messageId;

        @Nullable
        @SerializedName("attachment")
        public VKApiAttachments.Entry entry;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/response/AttachmentsHistoryResponse.kt
package dev.ragnarok.fenrir.api.model.response

import dev.ragnarok.fenrir.api.model.VKApiAttachments
import dev.ragnarok.fenrir.api.model.VKApiCommunity
import dev.ragnarok.fenrir.api.model.VKApiUser
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class AttachmentsHistoryResponse {
    @SerialName("items")
    var items: List<One>? = null

    @SerialName("next_from")
    var next_from: String? = null

    @SerialName("profiles")
    var profiles: List<VKApiUser>? = null

    @SerialName("groups")
    var groups: List<VKApiCommunity>? = null

    @Serializable
    class One {
        @SerialName("message_id")
        var messageId = 0

        @SerialName("attachment")
        var entry: VKApiAttachments.Entry? = null
    }
}